src/main/scala/aco/ACOSystem.scala

```diff
@@ -11,3 +11,7 @@ case class ACOSystem(
   nodes: List[ActorRef],
   config: ACOConfiguration)
 
+object ACOUtil {
+  val r = new scala.util.Random(System.currentTimeMillis)
```

Comment:
You could make this implicit if you want, and then just call nextDouble() when needed. Not too important though, it just makes the code cleaner